### PR TITLE
Improve handling of layerless images

### PR DESCRIPTION
**Pull request by nalind**
Update cache logic and image parent/child logic to recognize images that don't include any layers.

### CODE REVIEW

**Comment by giuseppe** [comment]
LGTM

File: cmd/buildah/common.go [hunk 1]
```diff
@@ -271,9 +271,12 @@ func getParent(ctx context.Context, sc *types.SystemContext, store storage.Store
 	if err != nil {
 		return nil, errors.Wrapf(err, "unable to retrieve image list from store")
 	}
-	childLayer, err := store.Layer(child.TopLayer)
-	if err != nil {
-		return nil, errors.Wrapf(err, "unable to retrieve layer list from store")
+	var childLayer *storage.Layer
```
Comment (TomSweeneyRedHat) [inline cmd/buildah/common.go]: Just to be completely clear, could I suggest childLayer to childTopLayer?

File: cmd/buildah/common.go [hunk 2]
```diff
@@ -331,12 +334,15 @@ func getChildren(ctx context.Context, sc *types.SystemContext, store storage.Sto
 		if child.ID == parent.ID {
 			continue
 		}
-		childLayer, err := store.Layer(child.TopLayer)
-		if err != nil {
-			return nil, errors.Wrapf(err, "unable to retrieve information about layer %q from store", child.TopLayer)
-		}
-		if childLayer.Parent != parent.TopLayer && childLayer.ID != parent.TopLayer {
-			continue
+		var childLayer *storage.Layer
```
Comment (TomSweeneyRedHat) [inline cmd/buildah/common.go]: DItto change this var name too if you touch up the other.

File: imagebuildah/build.go [hunk 3]
```diff
@@ -1219,15 +1219,18 @@ func (s *StageExecutor) layerExists(ctx context.Context, currNode *parser.Node,
 		return "", errors.Wrap(err, "error getting image list from store")
 	}
 	for _, image := range images {
-		layer, err := s.executor.store.Layer(image.TopLayer)
-		if err != nil {
-			return "", errors.Wrapf(err, "error getting top layer info")
+		var layer *storage.Layer
```
Comment (TomSweeneyRedHat) [inline imagebuildah/build.go]: suggest image to imageTopLayer here too.

**Comment by TomSweeneyRedHat** [comment]
LGTM
thanks for the var name touch ups @nalind

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Comment by rhatdan** [comment]
@rh-atomic-bot retry

**Comment by rhatdan** [comment]
@rh-atomic-bot retry

**Pull request closed without merging**